include/boost/process/v2/impl/exe.ipp

```diff
@@ -0,0 +1,313 @@
+{
+    std::boost::filesystem::path;
+    char exe[PATH_MAX];
+    if (realpath(("/proc/" + std::to_string(pid) + "/exe").c_str(), exe)) 
+    {
+        path = exe;
+    }
+    if (path.string().empty())
+    {
+        ec = detail::get_last_error();
+    }
+    return path;
+}
+
+#elif defined(__FreeBSD__) || defined(__DragonFly__)
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec) 
+{
+    std::boost::filesystem::path;
+    int mib[4]; 
+    std::size_t len;
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PATHNAME;
+    mib[3] = pid;
+    if (sysctl(mib, 4, nullptr, &len, nullptr, 0) == 0) 
+    {
+        std::string strbuff;
+        strbuff.resize(len, '\0');
+        char *exe = strbuff.data();
+        if (sysctl(mib, 4, exe, &len, nullptr, 0) == 0) 
+        {
+            char buffer[PATH_MAX];
+            if (realpath(exe, buffer)) 
+            {
+                path = buffer;
+            }
+        }
+    }
+    if (path.string().empty())
+    {
+        ec = detail::get_last_error();
+    }
+    return path;
+}
+
+#elif defined(__NetBSD__)
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec)
+{
+    std::boost::filesystem::path;
+    int mib[4]; 
+    std::size_t len;
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC_ARGS;
+    mib[2] = pid;
+    mib[3] = KERN_PROC_PATHNAME;
+    if (sysctl(mib, 4, nullptr, &len, nullptr, 0) == 0)
+    {
+        std::string strbuff;
+        strbuff.resize(len, '\0');
+        char *exe = strbuff.data();
+        if (sysctl(mib, 4, exe, &len, nullptr, 0) == 0)
+        {
+            char buffer[PATH_MAX];
+            if (realpath(exe, buffer))
+            {
+                path = buffer;
+            }
+        }
+    }
+    if (path.string().empty())
+    {
+        ec = detail::get_last_error();
+    }
+    return path;
+}
+
+#elif defined(__OpenBSD__)
+
+boost::filesystem::path exe_path(pid_type pid, error_code & ec)
+{
+    std::boost::filesystem::path;
+    auto is_executable = [](pid_type pid, std::string in, std::string *out) 
+    {
+        *out = "";
+        bool success = false;
+        struct stat st;
+        if (!stat(in.c_str(), &st) && (st.st_mode & S_IXUSR) && (st.st_mode & S_IFREG)) 
+        {
+            char executable[PATH_MAX];
+            if (realpath(in.c_str(), executable)) 
+            {
+                int cntp = 0;
+                kinfo_file *kif = nullptr;
+                kd = kvm_openfiles(nullptr, nullptr, nullptr, KVM_NO_FILES, nullptr);
+                if (!kd) return false;
+                if ((kif = kvm_getfiles(kd, KERN_FILE_BYPID, pid, sizeof(struct kinfo_file), &cntp))) 
+                {
+                    for (int i = 0; i < cntp; i++) 
+                    {
+                        if (kif[i].fd_fd == KERN_FILE_TEXT) 
+                        {
+                            if (st.st_dev == (dev_t)kif[i].va_fsid || st.st_ino == (ino_t)kif[i].va_fileid) 
```

Comment:
that's one `if` statement.

`static_cast` plz

Reply:
you mean combine the fd if statement with dev/ino or split dev/ino into two different if statements?